internal/output/summary/data.go

```diff
@@ -393,3 +404,28 @@ func metricValueGetter(summaryTrendStats []string) func(metrics.Sink, time.Durat
 		return result
 	}
 }
+
+var percentileThresholdSourceRe = regexp.MustCompile(`^p\((\d+(?:\.\d+)?)\)\s*([<>=])`)
+
+func extractPercentileThresholdSource(source string) (agg string, percentile float64, isPercentile bool) {
```

Comment:
I know this isn't ideal, but I cannot see any alternative to this that doesn't imply exposing more things from the `metrics` package and type, and I think that isn't ideal as well because we do plan to do some refactors there soon-ish. 

So, that's why I suggest to go with this, which should be fine as I don't expect the thresholds format to change anytime soon, especially now with v1.0 and the versioning strategy and guarantees.